runtime/compiler/x/codegen/X86PrivateLinkage.cpp

```diff
@@ -2377,23 +2359,20 @@ TR::Register *TR::X86PrivateLinkage::buildCallPostconditions(TR::X86CallSite &si
             }
          }
 
-      if (cg()->useSSEForSinglePrecision() || cg()->useSSEForDoublePrecision())
+      TR_LiveRegisters *lr = cg()->getLiveRegisters(TR_FPR);
+      if(!lr || lr->getNumberOfLiveRegisters() > 0)
          {
-         TR_LiveRegisters *lr = cg()->getLiveRegisters(TR_FPR);
-         if(!lr || lr->getNumberOfLiveRegisters() > 0)
+         for (regIndex = TR::RealRegister::FirstXMMR; regIndex <= TR::RealRegister::LastXMMR; regIndex = (TR::RealRegister::RegNum)(regIndex + 1))
             {
-            for (regIndex = TR::RealRegister::FirstXMMR; regIndex <= TR::RealRegister::LastXMMR; regIndex = (TR::RealRegister::RegNum)(regIndex + 1))
+            TR_ASSERT(regIndex != highReturnRegIndex, "assertion failure");
```

Comment:
While you are here can you fix the assert message please?

Reply:
I've updated the assert message; though I'm not sure if this assertion has any meaningful usage here. `highReturnRegIndex` is the register for returning high 32 bits of a 64-bit primitive on 32-bit ISA. It should only be a GPR and hence not an XMM register.

Reply:
Do we need a better assert?

Reply:
Probably not. This loop is to kill XMM registers that are neither the return register nor any of the preserved registers. I guess the original author wanted to be defensive so that it can detect a misused "high return register"?